netbox/utilities/tests/test_jobs.py

```diff
@@ -0,0 +1,134 @@
+
+
+class EnqueueTest(BackgroundJobTestCase):
+    """
+    Test enqueuing of `BackgroundJob`.
+    """
+
+    def test_enqueue(self):
+        instance = Job()
+        for i in range(1, 3):
+            job = TestBackgroundJob.enqueue(instance, schedule_at=self.get_schedule_at())
+
+            self.assertIsInstance(job, Job)
+            self.assertEqual(TestBackgroundJob.get_jobs(instance).count(), i)
+
+    def test_enqueue_once(self):
+        job = TestBackgroundJob.enqueue_once(instance=Job(), schedule_at=self.get_schedule_at())
+
+        self.assertIsInstance(job, Job)
+        self.assertEqual(job.name, TestBackgroundJob.__name__)
+
+    def test_enqueue_once_twice_same(self):
+        instance = Job()
+        schedule_at = self.get_schedule_at()
+        job1 = TestBackgroundJob.enqueue_once(instance, schedule_at=schedule_at)
+        job2 = TestBackgroundJob.enqueue_once(instance, schedule_at=schedule_at)
+
+        self.assertEqual(job1, job2)
+        self.assertEqual(TestBackgroundJob.get_jobs(instance).count(), 1)
+
+    def test_enqueue_once_twice_different_schedule_at(self):
+        instance = Job()
+        job1 = TestBackgroundJob.enqueue_once(instance, schedule_at=self.get_schedule_at())
+        job2 = TestBackgroundJob.enqueue_once(instance, schedule_at=self.get_schedule_at(2))
+
+        self.assertNotEqual(job1, job2)
+        self.assertRaises(Job.DoesNotExist, job1.refresh_from_db)
+        self.assertEqual(TestBackgroundJob.get_jobs(instance).count(), 1)
+
+    def test_enqueue_once_twice_different_interval(self):
+        instance = Job()
+        schedule_at = self.get_schedule_at()
+        job1 = TestBackgroundJob.enqueue_once(instance, schedule_at=schedule_at)
+        job2 = TestBackgroundJob.enqueue_once(instance, schedule_at=schedule_at, interval=60)
+
+        self.assertNotEqual(job1, job2)
+        self.assertEqual(job1.interval, None)
+        self.assertEqual(job2.interval, 60)
+        self.assertRaises(Job.DoesNotExist, job1.refresh_from_db)
+        self.assertEqual(TestBackgroundJob.get_jobs(instance).count(), 1)
+
+    def test_enqueue_once_with_enqueue(self):
+        instance = Job()
+        job1 = TestBackgroundJob.enqueue_once(instance, schedule_at=self.get_schedule_at(2))
+        job2 = TestBackgroundJob.enqueue(instance, schedule_at=self.get_schedule_at())
+
+        self.assertNotEqual(job1, job2)
+        self.assertEqual(TestBackgroundJob.get_jobs(instance).count(), 2)
+
+    def test_enqueue_once_after_enqueue(self):
+        instance = Job()
+        job1 = TestBackgroundJob.enqueue(instance, schedule_at=self.get_schedule_at())
+        job2 = TestBackgroundJob.enqueue_once(instance, schedule_at=self.get_schedule_at(2))
+
+        self.assertNotEqual(job1, job2)
+        self.assertRaises(Job.DoesNotExist, job1.refresh_from_db)
+        self.assertEqual(TestBackgroundJob.get_jobs(instance).count(), 1)
+
+    def test_enqueue_system(self):
```

Comment:
Would be good to add a test with the name parameter.

Reply:
@arthanson can you elaborate on this please? Passing `name` in `enqueue()` and `enqueue_once()` is not supported as documented. Test for automatic setting of the job's `name` are implemented in `BackgroundJobTest`.

Reply:
Partly it was used in runscript but I guess that might have been a mistake as it looks like it is removed now, but I'm not understanding why it is a parameter if it can't be passed in?

Reply:
One can pass a `name` to a `Job` object when using it standalone. `BackgroundJob` augments `Job` and sets the `name` parameter automatically, so it must not be set when calling `BackgroundJob.enqueue()`.

I forgot removing the parameter in `runscript`. This was fixed with ecf8e79fddcfd5b885fac3c29231edbcd73a12d8.